react-project/src/reducers/treemapSlice.js

```diff
@@ -1,73 +1,104 @@
+  "bytes": 1,
+  "busFactorStatus": {
+    "busFactor": 1
   },
-  simulation: {
-    isSimulationMode: false,
-    lastUsedRemovedAuthorsList: [],
-    miniTreemap: {
+  "users": [
+    {
+      "email": "place.holder@mail.com",
+      "authorship": 0.5
+    },],
+  "children": [
+    {
+      "name": "place_holder",
+      "path": "/place_holder",
+      "bytes": 1,
+      "busFactorStatus": {
+        "old": true
+      }
+    },]
+}
+
+function convertTreeToState(tree) {
+  return {
+    tree: tree,
+    mainTreemap: {
+      currentStatsPath: tree.path,
+      currentVisualizationPath: tree.path,
+      ignored: [],
+      isRecalculationEnabled: false,
       previousPathStack: [],
-      previousVisualizationData: [],
-      visualizationData: initialMiniTreeMapData,
-      visualizationPath: fullData.path,
     },
-    removedAuthors: [],
-  },
-  filters: [],
-};
+    simulation: {
+      isSimulationMode: false,
+      lastUsedRemovedAuthorsList: [],
+      miniTreemap: {
+        previousPathStack: [],
+        previousVisualizationData: [],
+        visualizationPath: tree.path,
+      },
+      removedAuthors: [],
+    },
+    filters: [],
+  };
+}
 
-function getDataWithPathQuery(fullData, pathQuery, developersToRemove) {
-  let newData = jp.query(fullData, pathQuery);
-  let result = calculateBusFactor(newData[0], developersToRemove);
 
-  return result;
+function goThroughTree(tree, path) {
+  if (path === ".") return tree
+
+  const parts = path.split('/')
+  let node = tree
+  for (let i = 0; i < parts.length; i++) {
+    const part = parts[i]
+    if (i === 0 && part === "") continue
+    node = node.children.filter((it) => it.name === part)[0]
+  }
+  return node
 }
 
-function initializeBusFactorDeltaProperties(dataRootNode) {
-  if (dataRootNode == null) throw new Error("Empty data file");
+function getDataRecalculculated(fullData, pathQuery, developersToRemove) {
+  let newData = goThroughTree(fullData, pathQuery);
+  return calculateBusFactor(newData, developersToRemove);
+}
 
-  if (!("busFactorStatus" in dataRootNode)) {
-    dataRootNode.busFactorStatus = {};
-  }
-  Object.defineProperties(dataRootNode.busFactorStatus, {
-    nodeStatus: {
-      value: "original",
-      writable: true,
-    },
-    delta: {
-      value: 0,
-      writable: true,
-    },
-  });
-  // dataRootNode.busFactorStatus.nodeStatus = "original";
-  // dataRootNode.busFactorStatus.delta = 0;
+export function initializeBusFactorDeltaProperties(node) {
+  if (node == null) throw new Error("Empty data file");
 
-  if (dataRootNode.children) {
-    for (let count = 0; count < dataRootNode.children.length; count++) {
-      dataRootNode.children[count] = initializeBusFactorDeltaProperties(
-        dataRootNode.children[count]
-      );
+  if (node.children) {
+    return {
+      ...node,
+      children: node.children.map((it) => {
+        const result =
+          Object.fromEntries(Object.entries(it).filter(e => e[0] !== 'children'))
```

Comment:
Does this line grab all properties named anything other than 'children' and assigns them a busFactor property with a value of {}